tests/test_encode.py

```diff
@@ -5,40 +5,57 @@
 
-params = {"cat_cols": [], "ord_dict": ord_dict}
 
-
-class TestOrdinalEncoding:
+class TestEncoding:
     # test for empty input
     def test_empty_df(self):
+        params = {"target_label": "Price", "ord_dict": ord_dict}
         with pytest.raises(ValueError):
-            encoder = EncodeData(target_label="Price", params=params)
-            encoder.encode()
+            encoder = EncodeData()
+            encoder.encode(params=params)
 
     # test for warning
     def test_warning(self):
         train_csv = pd.read_csv("datasets/encoding/testnew.csv")
+        params = {"train_df": train_csv, "ord_dict": ord_dict}
         with pytest.warns(UserWarning):
-            encoder = EncodeData(train_df=train_csv, params=params)
-            encoder.encode()
+            encoder = EncodeData()
+            encoder.encode(params=params)
 
     # test ordinal encoding
     def test_mapping(self):
         train_csv = pd.read_csv("datasets/encoding/testnew.csv")
-        encoder = EncodeData(train_df=train_csv, target_label="Price", params=params)
-        train = encoder.encode()
-        assert train[0]["ProfessionEncoded"].nunique() == 3
-        assert train[0]["ProfessionEncoded"][2] == 3
+        params = {"train_df": train_csv, "target_label": "Price", "ord_dict": ord_dict}
+        encoder = EncodeData()
+        encoder.encode(params=params)
+        assert params["train_df"]["ProfessionEncoded"].nunique() == 3
+        assert params["train_df"]["ProfessionEncoded"][2] == 3
         assert Counter(params["ord_dict"]["Profession"].values()) == Counter(
-            train[0]["ProfessionEncoded"].unique()
+            params["train_df"]["ProfessionEncoded"].unique()
         )
 
     # test for empty mapping
     def test_empty_weight_mapping(self):
         train_csv = pd.read_csv("datasets/encoding/testnew.csv")
+        ord_dict1 = ord_dict.copy()
+        ord_dict1["Size"] = None
+        params = {"train_df": train_csv, "target_label": "Price", "ord_dict": ord_dict1}
         with pytest.raises(ValueError):
-            params["ord_dict"]["Size"] = None
-            encoder = EncodeData(
-                train_df=train_csv, target_label="Price", params=params
-            )
-            encoder.encode()
+            encoder = EncodeData()
+            encoder.encode(params=params)
+
+    # test for one-hot encoding
+    def test_one_hot_encoding(self):
+        train_csv = pd.read_csv("datasets/encoding/testnew.csv")
+        params = {
+            "train_df": train_csv,
+            "target_label": "Price",
+            "cat_cols": ["Professions", "Test", "Labels"],
+            "ord_dict": ord_dict,
+            "one_hot": True,
+        }
+        encoder = EncodeData()
+        encoder.encode(params=params)
+        assert "Test_Tata" in params["train_df"].columns
+        assert params["train_df"]["Test_Tata"][1] == 1
+        assert "Profession_HOD" not in params["train_df"].columns
```

Comment:
Can you put this in a separate test? It was kind of hard to spot what assertion implies.